packages/shared/src/components/post/GoBackHeaderMobile.tsx

```diff
@@ -24,11 +25,16 @@ export function GoBackHeaderMobile({
   className,
 }: PropsWithChildren<WithClassNameProps>): ReactElement {
   const router = useRouter();
+  const goHome = useCallback(() => router.push('/'), [router]);
+  const logoButton = (
+    <Logo onLogoClick={goHome} position={LogoPosition.Initial} />
+  );
+
   const canGoBack =
-    !!globalThis?.history?.length && (checkSameSite() || isDevelopment);
+    globalThis?.history?.length > 1 && (checkSameSite() || isDevelopment);
```

Comment:
One thing actually, why are we checking here for >1 ?

Reply:
Because when you land on a page, the history is already length 1 -> the page you have just opened in the browser. So previous check did not make sense, since it would be always true.